rpc/client.go

```diff
@@ -101,11 +103,20 @@ type response[T serde.ResponsePayload] struct {
 // NewRpcClient creates a new RpcClient
 func NewRpcClient(trans transport.Requester) (RpcClientApi, error) {
 	ctx, cancel := context.WithCancel(context.Background())
-
-	c := &rpcClient{trans, &safesync.Map[chan struct{}]{}, &safesync.Map[chan query.LedgerChannelInfo]{}, &safesync.Map[chan query.PaymentChannelInfo]{}, cancel, &sync.WaitGroup{}, common.Address{}, slog.Default()}
+	c := &rpcClient{
+		transport:             trans,
+		completedObjectives:   &safesync.Map[chan struct{}]{},
+		ledgerChannelUpdates:  &safesync.Map[chan query.LedgerChannelInfo]{},
+		paymentChannelUpdates: &safesync.Map[chan query.PaymentChannelInfo]{},
+		cancel:                cancel,
+		wg:                    &sync.WaitGroup{},
```

Comment:
Slightly out of scope, but I wonder if we should rename this wait group (the one you added has a good name).